src/components/list/list-renderer.tsx

```diff
@@ -143,13 +144,25 @@ export class ListRenderer {
                 {...attributes}
             >
                 {item.icon ? this.renderIcon(this.config, item) : null}
+                {this.getCustomComponent(item)}
                 {this.renderText(item)}
                 {this.twoLines && this.avatarList ? this.renderDivider() : null}
                 {this.renderActionMenu(item.actions)}
             </li>
         );
     };
 
+    private getCustomComponent(item: ListItem): Element {
+        if (!item?.component) {
+            return;
+        }
+
+        const WebComponent = item?.component?.name;
+        const props = item?.component?.props;
+
+        return <WebComponent {...props} />;
```

Comment:
We probably need to set some restrictions here. Right now there's nothing stopping me from inserting a _huge_ component for an item that will break the whole list. I think we need a wrapper element here that sets a max size of the component

Reply:
I don't think so. It should be totally up to the consumer to make this as big as they want. Or even maybe let it take the entire width of the list item. We never know the use case